TASK-57050: No notification is sent to the mentioned user when the mention is done from the edition form

IlhemEssaadi opened the pull request:
ISSUES : When updating an article and mentioning a user, no notification is sent to the mentioned user.
FIX : The problem is that the mentioned users have been removed from the body of the updated news before sending them in the notification. We fixed that by adding to the news object an originalBody attribute containing the body of the original news from  which we get the previous mentions and remove them from the updated news.

azayati commented on services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImpl.java:
```diff
@@ -173,14 +172,7 @@ public News updateNews(News news, String updater, Boolean post, boolean publish)
-            newMentionedNews.setBody(newMentionedNews.getBody().replaceAll("@"+username, ""));
-          });
-        }
-        sendNotification(updater, newMentionedNews, NotificationConstants.NOTIFICATION_CONTEXT.MENTION_IN_NEWS);
```
Normally you should keep this treatment in order to not send additional notifications to the users already mentioned.
The problem is around this instruction `Set<String> previousMentions = NewsUtils.processMentions(news.getBody());`

- Here news is the new updated news object

You should get instead the old news object with `newsStorage.getNewsById(news.getId(), false)`

cc @boubaker

azayati reviewed (changes requested):
see review

azayati commented on services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImpl.java:
```diff
@@ -151,7 +151,8 @@ public News updateNews(News news, String updater, Boolean post, boolean publish)
       throw new IllegalArgumentException("User " + updater + " is not authorized to update news");
     }
-    Set<String> previousMentions = NewsUtils.processMentions(news.getBody());
+    News previousNews = newsStorage.getNewsById(news.getId(), false);
```
```suggestion
    News originalNews = newsStorage.getNewsById(news.getId(), false);
```

azayati reviewed (changes requested):
see review

IlhemEssaadi committed:
TASK-57050: Fix no notification is sent to the mentioned user when the mention is done from the edition form

ISSUES : when update an article and mention a user, no notification is sent to the mentioned user.
FIX : the problem is that in the case of update and when we mention user the mentioned user was removed from the body of the news before sending it in the notification, so we can't get  mentionedIds from the contentBody. it fixed by sending the BodyContent news with the mentioned users.

azayati reviewed (approved):
LGTM
